src/utils/castFilter.ts

```diff
@@ -17,7 +17,47 @@ export default function castFilter<T extends ADPBaseModel>(
       obj[key] = path(key).cast(obj[key], null, null);
     } catch (e) {}
 
-    if (allowedRegexes.includes(key) && typeof obj[key] === "string") {
+    /**  Parse MongoDB Query Operators **/
+    let splittedKey = key.split("_");
+    if (
+      splittedKey.length > 1 &&
+      ["eq", "gt", "gte", "in", "lt", "lte", "ne", "nin"].includes(
+        splittedKey[splittedKey.length - 1]
+      )
+    ) {
+      let operator = splittedKey.pop();
+      let field = splittedKey.join("_");
+
+      obj[field] =
+        obj[field] && typeof obj[field] == "object" ? obj[field] : {};
+      switch (operator) {
+        case "eq":
+          obj[field].$eq = obj[key];
```

Comment:
I do not think it is wise to instantly pass it like such. It may cause unwanted issues, such as being able to pass arrays and objects unintentionally into the $eq, $gt, etc. fields.

Reply:
Also, things like $nin, $in seems to receive arrays. Regular strings should be put into an array consisting of that one string, as the obj isn't guaranteed to be an array when it comes in.

Reply:
> Also, things like $nin, $in seems to receive arrays. Regular strings should be put into an array consisting of that one string, as the obj isn't guaranteed to be an array when it comes in.

`$in` and `$nin` automatically parse single values into an array. There's no need to do this manually.
Example:
`GET /api/users?name_in=paulr113`
Results in:
`users.find({ name: { '$in': [ 'paulr113' ] } }, { limit: 25, projection: {} })`

Reply:
> I do not think it is wise to instantly pass it like such. It may cause unwanted issues, such as being able to pass arrays and objects unintentionally into the $eq, $gt, etc. fields.

How would you go about solving that?

Reply:
Great question, I'm actually not too sure either. May need something to validate all the types passed into the queries.